### PR TITLE
Mopup for aspnetcore-tooling migration.

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Following up to https://github.com/dotnet/aspnetcore/pull/21581/. We now have changes up to https://github.com/dotnet/aspnetcore-tooling/commit/1b16a37fc76ff0cc2b7a02486fb3eb0289f41738 migrated to aspnetcore.

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]

**Review by ryanbrandenburg — APPROVED** [review]

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 1]
```diff
@@ -25,9 +25,11 @@
     <Reference Include="Microsoft.Build.Framework" />
     <Reference Include="Microsoft.Build.Utilities.Core" />
     <Reference Include="System.Reflection.Metadata"  Condition="'$(TargetFramework)'=='net46'" />
-
-    <Reference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions" ReferenceOutputAssembly="false" Condition="'$(TargetFramework)'=='$(DefaultNetCoreTargetFramework)'" />
     <Reference Include="rzc" ReferenceOutputAssembly="false" Condition="'$(TargetFramework)'=='$(DefaultNetCoreTargetFramework)'" />
+
+    <ProjectReference Include="..\..\Microsoft.AspNetCore.Mvc.Razor.Extensions\src\Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj" ReferenceOutputAssembly="false" >
+      <SetTargetFramework>TargetFramework=netstandard2.0</SetTargetFramework>
+    </ProjectReference>
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:32]: Why switch from `<Reference/>` to `<ProjectReference/>`? The `%(SetTargetFramework)` and `%(ReferenceOutputAssembly)` metadata won't be lost when using `<Reference/>`. Let's minimize the special cases.

Comment (JunTaoLuo) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:32]: We seemed to use this pattern else where and I didn't see a <Reference> element using SetTargetFramework in our repo.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 2]
```diff
@@ -25,9 +25,11 @@
     <Reference Include="Microsoft.Build.Framework" />
     <Reference Include="Microsoft.Build.Utilities.Core" />
     <Reference Include="System.Reflection.Metadata"  Condition="'$(TargetFramework)'=='net46'" />
-
-    <Reference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions" ReferenceOutputAssembly="false" Condition="'$(TargetFramework)'=='$(DefaultNetCoreTargetFramework)'" />
     <Reference Include="rzc" ReferenceOutputAssembly="false" Condition="'$(TargetFramework)'=='$(DefaultNetCoreTargetFramework)'" />
+
+    <Reference Include="..\..\Microsoft.AspNetCore.Mvc.Razor.Extensions\src\Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj" ReferenceOutputAssembly="false" >
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:30]:
```suggestion
    <Reference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions" ReferenceOutputAssembly="false" >
```

**Comment by dougbu** [comment]
> We seemed to use this pattern else where and I didn't see a element using SetTargetFramework in our repo.

Can't continue the conversation now that you've addressed my comment. FYI `%(SetTargetFramework)` metadata is an input to the usual SDK logic for building project references.

**Comment by JunTaoLuo** [comment]
@dougbu I'm running into some issues with your suggestion of using `SetTargetFramework`. Since it's quite urgent to fix the build, I'm just going to use my original fix.

**Pull request merged**